### PR TITLE
Hide 'Run Code' button for programming questions in manually-graded assessments unless necessary

### LINKED ISSUE #1793: Hide 'Run code' button when there are no test cases
Programming questions are not autogradeable when there are no test cases. In these situations, the 'Run code' button should be hidden from students.

Consider what to do if there are only evaluation tests. Staff should be able to run the tests, but students should see it as not autogradeable or it will be confusing.

### PR DESCRIPTION
**Pull request by weiqingtoh**
This PR introduces the following behavior for programming questions in manually-graded assessments:
 - Students see the button only if there are public or private test cases
 - Staff see the button if there are any test cases
 - Added a tooltip warning for staff (as they see the button, but not the student) when question only has evaluation test cases
 - Added feature specs to test for students and staff

Fixes #1793.

### CODE REVIEW

File: config/locales/en/course/assessment/answer/answers.yml [hunk 1]
```diff
@@ -10,6 +10,9 @@ en:
         manually_graded:
           run_code: 'Run Code'
           run_code_with_limit: 'Run ( %{limit} times left )'
+          run_code_evaluation_warning: >
+            You are able to see this because you are a staff and can view evaluation tests.
+            Note that students are not able to view the tests, nor this button.
```
Comment (fonglh) [inline config/locales/en/course/assessment/answer/answers.yml]: "Students cannot see these tests nor this button."

**Comment by fonglh** [comment]
Should we add the little privacy triangle inside the button and have the tooltip there instead? So it'll be similar to the warning flags for private and evaluation test cases.

image

Something like this?
image

**Comment by weiqingtoh** [comment]
I don't think there is a need for an extra icon because the evaluation tests have it already. Staff who click that button have to hover over and will definitely see the message. 

I prefer to be a little more verbose with the tooltip message; it seems like there was a misunderstanding between staff and student because the staff thought the student could see the tests.

**Comment by fonglh** [comment]
Oh I was just referring to changing the 2nd line. Keep the first one about being staff and being able to view the evaluation tests.

**Comment by weiqingtoh** [comment]
Oh then that works :+1:  will fix

File: spec/features/course/assessment/submission/manually_graded_spec.rb [hunk 2]
```diff
@@ -197,6 +215,15 @@
           to eq(edit_course_assessment_submission_path(course, assessment, submission))
         expect(submission.reload.published?).to be(true)
       end
+
+      scenario 'I can run code on autograded programming questions', js: true do
```
Comment (allenwq) [inline spec/features/course/assessment/submission/manually_graded_spec.rb]: Is `js: true` needed ?

Comment (weiqingtoh) [inline spec/features/course/assessment/submission/manually_graded_spec.rb]: 👍 Not sure why I didn't remove this.

**Comment by allenwq** [comment]
LGTM

**Pull request merged**